### PR TITLE
Remove MapSecurityZoneWithUrlmon method and related code

### LINKED ISSUE #6919: Use MapSecurityZoneWithoutUrlmon and remove MapSecurityZoneWithUrlmon and related native code
MapSecurityZoneWithUrlmon uses an unsupported API call (the dll Imports below),  we should remove the unsupported calls so we sue supported APIs to improve quality.

Use MapSecurityZoneWithoutUrlmon and remove MapSecurityZoneWithUrlmon and related native code

- [ ] Review MapSecurityZoneWithoutUrlmon for completeness
  - https://github.com/PowerShell/PowerShell/blob/6d7799b7665665056911aa46d28be0c58b4e5d87/src/System.Management.Automation/utils/ClrFacade.cs#L189
- [ ] remove MapSecurityZoneWithUrlmon
  - https://github.com/PowerShell/PowerShell/blob/6d7799b7665665056911aa46d28be0c58b4e5d87/src/System.Management.Automation/utils/ClrFacade.cs#L295
- [ ] remove dllimports and similar related to MapSecurityZoneWithUrlmon
  - https://github.com/PowerShell/PowerShell/blob/c1c5344a8897262433141ecbc13bb06ac2c4bbef/src/System.Management.Automation/utils/ClrFacade.cs#L456
  - https://github.com/PowerShell/PowerShell/blob/c1c5344a8897262433141ecbc13bb06ac2c4bbef/src/System.Management.Automation/utils/ClrFacade.cs#L464

### PR DESCRIPTION
**Pull request by rjmholt**
## PR Summary

Fixes #6919: Removes the `MapSecurityZoneWithUrlmon` method and the native interface it uses.

`MapSecurityZoneWithUrlmon` uses an unsupported API call, and this ensures we use a supported API to improve quality.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by anmenaga — CHANGES_REQUESTED** [review]
Neither "PR Summary" nor original issue have a description why this change is necessary.

**Comment by TravisEz13** [comment]
@anmenaga 
> Neither "PR Summary" nor original issue have a description why this change is necessary.

@rjmholt says in the summary 
> MapSecurityZoneWithUrlmon uses an unsupported API call, and this ensures we use a supported API to improve quality.

**Review by TravisEz13 — APPROVED** [review]

File: src/System.Management.Automation/utils/ClrFacade.cs [hunk 1]
```diff
@@ -139,19 +139,11 @@ internal static SecurityZone GetFileSecurityZone(string filePath)
             Diagnostics.Assert(Path.IsPathRooted(filePath), "Caller makes sure the path is rooted.");
             Diagnostics.Assert(Utils.FileExists(filePath), "Caller makes sure the file exists.");
             string sysRoot = System.Environment.GetEnvironmentVariable("SystemRoot");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/ClrFacade.cs]: This variable is not used anymore.

File: src/System.Management.Automation/utils/ClrFacade.cs [hunk 2]
```diff
@@ -448,60 +403,6 @@ private static class NativeMethods
             internal static extern uint GetACP();
 
             public const int S_OK = 0x00000000;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/ClrFacade.cs]: Is this one still used anywhere?

File: src/System.Management.Automation/utils/ClrFacade.cs [hunk 3]
```diff
@@ -448,60 +403,6 @@ private static class NativeMethods
             internal static extern uint GetACP();
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/ClrFacade.cs]: This function doesn't seem to be used anymore.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

**Review by anmenaga — APPROVED** [review]

**Comment by anmenaga** [comment]
@TravisEz13 "edited" after I posted the request. :)

**Comment by TravisEz13** [comment]
@anmenaga  I'm not sure you get notified of edits and the timing wasn't perfectly clear.

**Review by daxian-dbw — APPROVED** [review]

**Comment by rjmholt** [comment]
Looks like the test failure is `pwrshplugin.dll`:
```
  Describing JEA session Get-Help test
    [-] Error occurred in Describe block 9.84s
      ItemNotFoundException: Cannot find path 'C:\projects\powershell\src\powershell-win-core\bin\Release
      etcoreapp2.1\win7-x64\publish\pwrshplugin.dll' because it does not exist.
```

**Comment by rjmholt** [comment]
Not sure if that can be fixed in this PR

**Pull request merged**